torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -133,7 +140,15 @@ def _init_mp_dtypes(self) -> None:
                 f"FSDP expects uniform original parameter dtype but got {orig_dtypes}"
             )
         self._orig_dtype = next(iter(orig_dtypes))
-        self._param_dtype = self._orig_dtype
+        reduce_dtypes = {fsdp_param.reduce_dtype for fsdp_param in self.fsdp_params}
+        if len(reduce_dtypes) != 1:
+            # This can be relaxed if we issue one reduce-scatter per reduce
+            # dtype (but we would need a way for users to specify multiple
```

Comment:
🤔 the reason we only checks for reduce_dtype but not param_dtype is because we possibly need to allgather params with mixed dtype like fp8?